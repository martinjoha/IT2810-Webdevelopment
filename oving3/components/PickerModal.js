import React, {Component} from 'react';
import {
    Text,
    View,
    StyleSheet, Modal, Button, Dimensions
}
    from 'react-native';


export default class PickerModal extends Component {

    constructor(props) {
        super();
    }

    render() {
        return (
            <View style={styles.outerView}>
                <Text style={{textAlign: 'center', marginBottom: 15,}}>{this.props.header}</Text>
                <Text
                    style={styles.inputField}
                    onPress={this.props.openModal}>{this.props.value}</Text>
                <Modal
                    animationType="slide"
                    transparent={true}
                    visible={this.props.visible}
                    onDismiss={this.props.closeModal}
                    onRequestClose={this.props.closeModal}>
                    <View style={styles.outerModal}>
                        <View style={styles.innerModal}>
                            <Button title={"OK"} onPress={this.props.closeModal}/>
                            {this.props.picker}
                        </View>
                    </View>
                </Modal>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    outerView: {
        flex: 0.3,
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'flex-start'
    },
    outerModal: {
        flex: 1,
        justifyContent: "flex-end",
    },
    innerModal: {
        backgroundColor: "#ffffff",
        height: Dimensions.get("window").height * 0.35
    },
    inputField: {
        backgroundColor: "white",
        width: "90%",
        padding: "2%",
    }
});



