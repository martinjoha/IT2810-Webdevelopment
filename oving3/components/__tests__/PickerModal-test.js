import React from "react";
import "react-native"
import PickerModal from "../PickerModal";
import renderer from "react-test-renderer";

it('PickerModal renders correctly', async () => {
    const tree = renderer.create(<PickerModal closeModal={()=>{}} />).toJSON();
    expect(tree).toMatchSnapshot()
});