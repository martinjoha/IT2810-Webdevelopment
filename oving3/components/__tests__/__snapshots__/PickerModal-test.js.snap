// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerModal renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 0.3,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 15,
        "textAlign": "center",
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "white",
        "padding": "2%",
        "width": "90%",
      }
    }
  />
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onDismiss={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ffffff",
            "height": 466.9,
          }
        }
      >
        <View
          accessibilityComponentType="button"
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#007AFF",
                    "fontSize": 18,
                    "padding": 8,
                    "textAlign": "center",
                  },
                ]
              }
            >
              OK
            </Text>
          </View>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
